projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, OnInit, Inject, Input, ElementRef } from '@angular/core';
+import {BaseValueComponent} from '../../base-value.component';
+import { ReadTextValueAsHtml } from '@knora/api';
+import { FormControl, FormGroup, FormBuilder } from '@angular/forms';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'kui-text-value-as-html',
+  templateUrl: './text-value-as-html.component.html',
+  styleUrls: ['./text-value-as-html.component.scss']
+})
+export class TextValueAsHtmlComponent extends BaseValueComponent implements OnInit {
+
+  @Input() displayValue?: ReadTextValueAsHtml;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+
+  customValidators = [];
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder, private el: ElementRef,) {
+    super();
+  }
+
+  ngOnInit() {
+    // initialize form control elements
+    this.valueFormControl = new FormControl(null);
+
+    this.commentFormControl = new FormControl(null);
+
+    this.valueChangesSubscription = this.commentFormControl.valueChanges.subscribe(
+      data => {
+        this.valueFormControl.updateValueAndValidity();
+      }
+    );
+
+    this.form = this.fb.group({
+      textValue: this.valueFormControl,
+      comment: this.commentFormControl
+    });
+
+    this.resetFormControl();
+  }
+
+  getInitValue() {    
+    if (this.displayValue !== undefined) {
+      this.displayValue.html = '<p>This is a very simple HTML document with a <a href="http://rdfh.ch/c9824353ae06" class="kui-link">link</a></p>';
+      this.el.nativeElement.innerHTML = this.displayValue.html;
```

Comment:
How would the comment be shown?

Reply:
Or could this be injected in some section in the template? I guess we need to sanitise it.

Reply:
see https://angular.io/guide/security

Reply:
Ideally we could assign the html to the value `FormControl` and sanitize it so it's rendered properly in the template.